### PR TITLE
HSEARCH-859 Review names and composition of configuration properties

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-859

@gsmet Would you mind taking a look, please?

### CODE REVIEW

File: backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/index/IndexStatus.java [hunk 1]
```diff
@@ -4,15 +4,15 @@
  * License: GNU Lesser General Public License (LGPL), version 2.1 or later
  * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
  */
-package org.hibernate.search.backend.elasticsearch.cfg;
+package org.hibernate.search.backend.elasticsearch.index;
```
Comment (gsmet) [inline backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/index/IndexStatus.java]: I must admit I liked having everything config under a specific package.

File: documentation/src/main/asciidoc/backend-elasticsearch.asciidoc [hunk 2]
```diff
@@ -265,7 +265,7 @@ link:{elasticsearchDocUrl}/index-modules.html#_static_index_settings[set the pro
 
 Hibernate Search includes a feature named "index lifecycle management",
 where it will automatically create, validate, update, or drop an index on startup or shutdown.
-`hibernate.search.default.elasticsearch.index_schema_management_strategy CREATE` (default)
+`hibernate.search.default.elasticsearch.lifecycle.strategy CREATE` (default)
```
Comment (gsmet) [inline documentation/src/main/asciidoc/backend-elasticsearch.asciidoc]: So you're back to the old name?

Comment (yrodiere) [reply documentation/src/main/asciidoc/backend-elasticsearch.asciidoc]: `index_schema_management_strategy` was the old name. I had forgotten to update this section of the documentation.

**Comment by yrodiere** [comment]
Merged, thanks

**Pull request merged**